src/grt/src/fastroute/src/FastRoute.cpp

```diff
@@ -1219,6 +1219,8 @@ NetRouteMap FastRouteCore::run()
 
   net_eo_.clear();
 
+  updateDbCongestion();
```

Comment:
This is the wrong place to call the update; this is 
Incremental global routing cannot afford to call this extremely slow function every time it runs.
updateDbCongestion is called by GlobalRouter::globalRoute so it is already called "before exiting global routing". What call/command sequence are you using that bypasses the existing call?

Reply:
Is the issue just that the error causes the later call to `updateDbCongestion` not to be called? If so, couldn't this call be moved to line 1224, right before the error is thrown to ensure the congestion information is kept, but is not called twice? Or alternatively, expose this function so the `heat map` could call it "itself" when needed, thus avoiding calls to this function unless needed.

Reply:
Peter is correct; the error is thrown before updateDbCongestion gets called.
I moved the error reporting (which really shouldn't have been inside fastroute anyway) in PR 
 grt mv congestion error so db congestion gets updated #1500